corral/test/integration/test_fetch.pony

```diff
@@ -266,3 +266,30 @@ class TestFetchRemoteGits is UnitTest
           h.complete(ar.exit_code() == 0)
         end
       })
+
+
+class TestFetchBadGitReference is UnitTest
+  var data: (DataClone | DataNone) = DataNone
+
+  fun name(): String => "integration/fetch/bad-git-reference"
+
+  fun ref set_up(h: TestHelper val) ? =>
+    data = DataClone(h, "bad-git-reference")?
+
+  fun tear_down(h: TestHelper val) => data.cleanup(h)
+
+  fun apply(h: TestHelper) ? =>
+    h.long_test(10_000_000_000)
+    Execute(h,
+      recover [
+        "fetch"
+        "--verbose"
+        "--bundle_dir"; Data(h, "bad-git-reference")?.path
+      ] end,
+      {(h: TestHelper, ar: ActionResult)(data=data) =>
```

Comment:
I'd prefer to not see this tested in this fashion. Relying on the stderr output is incredibly brittle.

I'd like to see these modeled after the tests included in https://github.com/ponylang/corral/pull/126/files

Where a custom `CmdResultReceiver` is used. In this case of this PR it would be a custom RepoOperationResultReceiver.

Reply:
For me, the important part of that test is verifying the exit code is properly set to indicate an error. The checks on stdout/stderr can be removed if you want flexibility to change output without breaking tests.

I can switch it to the unit test form too but there's a bit of work to update the model to support asserting on `Context.Env.exitcode`. Also I don't think it's as good of a test since some other part of the Corral can change the exit code after `_Updater` finishes.

Reply:
If the important part is the exit code, then ya, let's remove the stdout/stderr checking.